docs/modules/ROOT/pages/api/erc20.adoc

```diff
@@ -141,52 +110,102 @@ Emitted when the allowance of a `spender` for an `owner` is set.
 
 [.hljs-theme-dark]
 ```javascript
-use openzeppelin::token::erc20::ERC20;
+use openzeppelin::token::erc20::interface::IERC20Metadata;
 ```
 
-Implementation of the <<IERC20,IERC20>> interface.
+IERC20Metadata interface as defined in {eip20}.
 
 [.contract-index]
-.Constructor
+.Functions
 --
-* xref:#ERC20-constructor[`++constructor(self, name, symbol, initial_supply, recipient)++`]
+* xref:#IERC20Metadata-name[`++name()++`]
+* xref:#IERC20Metadata-symbol[`++symbol()++`]
+* xref:#IERC20Metadata-decimals[`++decimals()++`]
 --
 
+[#IERC20Metadata-Functions]
+==== Functions
+
+[.contract-item]
+[[IERC20Metadata-name]]
+==== `[.contract-item-name]#++name++#++() → felt252++` [.item-kind]#external#
+
+Returns the name of the token.
+
+[.contract-item]
+[[IERC20Metadata-symbol]]
+==== `[.contract-item-name]#++symbol++#++() → felt252++` [.item-kind]#external#
+
+Returns the ticker symbol of the token.
+
+[.contract-item]
+[[IERC20Metadata-decimals]]
+==== `[.contract-item-name]#++decimals++#++() → u8++` [.item-kind]#external#
+
+Returns the number of decimals the token uses - e.g. `8` means to divide the token amount by `100000000` to get its user-readable representation.
+
+For example, if `decimals` equals `2`, a balance of `505` tokens should
+be displayed to a user as `5.05` (`505 / 10 ** 2`).
+
+Tokens usually opt for a value of `18`, imitating the relationship between
+Ether and Wei. This is the default value returned by this function, unless
+a custom implementation is used.
+
+NOTE: This information is only used for _display_ purposes: it in
+no way affects any of the arithmetic of the contract.
+
+[.contract]
+[[ERC20]]
+=== `++ERC20++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/token/erc20/erc20.cairo[{github-icon},role=heading-link]
+
+[.hljs-theme-dark]
+```javascript
+use openzeppelin::token::erc20::ERC20;
+```
+ERC20 component extending <<IERC20,IERC20>>.
+
 [.contract-index]
-.External functions
+.Embeddable functions
 --
 .ERC20Impl
-* xref:#ERC20-name[`++name(self)++`]
-* xref:#ERC20-symbol[`++symbol(self)++`]
-* xref:#ERC20-decimals[`++decimals(self)++`]
 * xref:#ERC20-total_supply[`++total_supply(self)++`]
 * xref:#ERC20-balance_of[`++balance_of(self, account)++`]
 * xref:#ERC20-allowance[`++allowance(self, owner, spender)++`]
 * xref:#ERC20-transfer[`++transfer(self, recipient, amount)++`]
 * xref:#ERC20-transfer_from[`++transfer_from(self, sender, recipient, amount)++`]
 * xref:#ERC20-approve[`++approve(self, spender, amount)++`]
 
-.Non-standard
+.ERC20MetadataImpl
+* xref:#ERC20-name[`++name(self)++`]
+* xref:#ERC20-symbol[`++symbol(self)++`]
+* xref:#ERC20-decimals[`++decimals(self)++`]
+
+.SafeAllowanceImpl
 * xref:#ERC20-increase_allowance[`++increase_allowance(self, spender, added_value)++`]
 * xref:#ERC20-decrease_allowance[`++decrease_allowance(self, spender, subtracted_value)++`]
+--
 
+[.contract-index]
+.camelCase support
```

Comment:
this is taking the impls below outside the "embeddable" section, breaking the grouping

Reply:
The idea was to break the grouping because I think it's difficult to follow having impls with both casings grouped together. Easy fix if we prefer it the other way. Or maybe change the section title to `Embeddable camelCase support/functions`?